src/scripts/blah.py

```diff
@@ -1,19 +1,44 @@
 """Common functions for BLAH python scripts"""
 
+import os
+import glob
+import sys
 from configparser import RawConfigParser
 from io import StringIO
 
 class BlahConfigParser(RawConfigParser, object):
 
-    def __init__(self, path='/etc/blah.config', defaults=None):
+    def __init__(self, path=None, defaults=None):
+        if path is None:
+            if "BLAHPD_CONFIG_LOCATION" in os.environ:
+                path = os.environ['BLAHPD_CONFIG_LOCATION']
+            elif "BLAHPD_LOCATION" in os.environ and os.path.isfile("%s/etc/blah.config" % os.environ['BLAHPD_LOCATION']):
+                path = "%s/etc/blah.config" % os.environ['BLAHPD_LOCATION']
+            elif "GLITE_LOCATION" in os.environ and os.path.isfile("%s/etc/blah.config" % os.environ['GLITE_LOCATION']):
+                path = "%s/etc/blah.config" % os.environ['GLITE_LOCATION']
+            else:
+                path = "/etc/blah.config"
         # RawConfigParser requires ini-style [section headers] but since
         # blah.config is also used as a shell script we need to fake one
+        config_dir = path + ".d/"
+        config_list = sorted(filter(os.path.isfile, glob.glob(config_dir + "*")))
+        config_list.insert(0, path)
+        user_config = os.path.expanduser("~/.blah/user.config")
+        if os.path.isfile(user_config):
+            config_list.append(user_config)
         self.header = 'blahp'
-        with open(path) as f:
-            config = f.read()
+        config = "[%s]\n" % (self.header)
+        for file in config_list:
+            with open(file) as f:
+                config += f.read()
```

Comment:
is it safe just to concatenate all the files together?

in particular i am worried about people who badly have a non-final file without a final newline at eof.

Would it be safer at least to ensure you don't join lines across files with something like this?

```suggestion
        for file in config_list:
            with open(file) as f:
                config += f.read() + "\n"
```